File: services/userJS/create.js
import { success, failure } from "../../libs/responseLib";
import {extractIdentityIdFromEvent} from "../../libs/dataExtraction";
import dynamoDb from "../../libs/dynamodb-lib";

export async function handler(event, context) {
  try {
    const data = typeof event.body === typeof "string" ? JSON.parse(event.body):event.body;
    const userId = data.userId || extractIdentityIdFromEvent(event);
    const params = {
      TableName: process.env.tableName,
      Item: {
        pK: `USER#${userId}`,
        sK: "USER",
        userName: data.userName,
        userMail: data.userMail,
        userPhone: data.userPhone,
        updatedAt: null,
        createdAt: Date.now()
      }
    };
    await dynamoDb.put(params);
    return success(params.Item);
  } catch (e) {
    return failure({ message: e.message });
  }
}

File: services/usersPy/create.py
from libs.responseLib import success, failure;
from libs.dataExtraction import extractUserSubFromEvent;
import boto3;
from os import environ;
from datetime import datetime
dynamodb = boto3.resource('dynamodb', region_name=environ.get('REGION'));
table = dynamodb.Table(environ.get('tableName'));
def handler(event, context):
    try:
        _event = json.load(event)
    except Exception as e:
        _event = event
    userId = _event['pathParameters']['userId'] if _event['pathParameters'] and _event['pathParameters']['userId'] else extractUserSubFromEvent(_event)
    try:
        item = {
            'pK': f"USER#{userId}",
            'sK': "USER",
            'userName': data.userName,
            'userMail': data.userMail,
            'userPhone': data.userPhone,
            'updatedAt': None,
            'createdAt': int(datetime.now().timestamp()*1000)
        }
        table.create_item(
            Item=item
        )
        return success(item);
    except Exception as e:
        return failure({'message': str(e)});
